plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobStore.java

```diff
@@ -105,6 +105,16 @@ class S3BlobStore implements BlobStore {
         this.priorityExecutorBuilder = priorityExecutorBuilder;
     }
 
+    @Override
+    public boolean isReloadable() {
+        return true;
+    }
+
+    @Override
+    public void reload(RepositoryMetadata repositoryMetadata) {
+        this.repositoryMetadata = repositoryMetadata;
```

Comment:
we will need to reload all the values which are initialized in the constructor .

Reply:
The other members being initialized in the constructor are not using the repository metadata. So reloading them or not won't have any effect:

```java
    S3BlobStore(
        S3Service service,
        S3AsyncService s3AsyncService,
        boolean multipartUploadEnabled,
        String bucket,
        boolean serverSideEncryption,
        ByteSizeValue bufferSize,
        String cannedACL,
        String storageClass,
        RepositoryMetadata repositoryMetadata,
        AsyncTransferManager asyncTransferManager,
        AsyncExecutorContainer priorityExecutorBuilder,
        AsyncExecutorContainer normalExecutorBuilder
    ) {
        this.service = service;
        this.s3AsyncService = s3AsyncService;
        this.multipartUploadEnabled = multipartUploadEnabled;
        this.bucket = bucket;
        this.serverSideEncryption = serverSideEncryption;
        this.bufferSize = bufferSize;
        this.cannedACL = initCannedACL(cannedACL);
        this.storageClass = initStorageClass(storageClass);
        this.repositoryMetadata = repositoryMetadata;
        this.asyncTransferManager = asyncTransferManager;
        this.normalExecutorBuilder = normalExecutorBuilder;
        this.priorityExecutorBuilder = priorityExecutorBuilder;
    }
```

Resolving this thread. Please re-open if needed.

Reply:
`bufferSize`, `serverSideEncryption` , `cannedACL` etc are using repo metadata only.

Reply:
Updated